CMake/HighFiveTargetDeps.cmake

```diff
@@ -0,0 +1,58 @@
+# Link against target system libs
+# -------------------------------
+
+# therefore making it possible to have new dependencies each build
+if(NOT TARGET highfive_deps)
+  add_library(highfive_deps INTERFACE)
+else()
+# Reset if imported
+set_target_properties(highfive_deps PROPERTIES
+  INTERFACE_COMPILE_DEFINITIONS ""
+  INTERFACE_COMPILE_OPTIONS ""
+  INTERFACE_INCLUDE_DIRECTORIES ""
+  INTERFACE_LINK_LIBRARIES ""
+  INTERFACE_LINK_OPTIONS ""
+  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES ""
+)
+endif()
+
+
+# HDF5
+if(NOT DEFINED HDF5_C_LIBRARIES)
+  set(HDF5_PREFER_PARALLEL ${HIGHFIVE_PARALLEL_HDF5})
+  find_package(HDF5 REQUIRED)
+endif()
+
+if(HIGHFIVE_PARALLEL_HDF5 AND NOT HDF5_IS_PARALLEL)
+  message(WARNING "Parallel HDF5 requested but libhdf5 doesnt support it")
+endif()
+
+target_include_directories(highfive_deps SYSTEM INTERFACE ${HDF5_INCLUDE_DIRS})
+target_link_libraries(highfive_deps INTERFACE ${HDF5_C_LIBRARIES})
+target_compile_definitions(highfive_deps INTERFACE ${HDF5_DEFINITIONS})
+
+# Boost
+if(USE_BOOST)
+  find_package(Boost REQUIRED COMPONENTS system serialization)
+  target_link_libraries(highfive_deps INTERFACE Boost::boost Boost::serialization)
+  target_include_directories(highfive_deps INTERFACE ${Boost_INCLUDE_DIR})
```

Comment:
If I'm not incorrect `Boost::boost` takes care of the include. I don't think there is need for the `target_include_directories` (but it would be good to check)

Reply:
I reverted to not using boost targets for support for cmake < 3.5.
Next main release we bump cmake requirements and start using targets.

Reply:
I guess that is fine. 

One could now have both implementations based on CMake version. Those of use using a new CMake version and that want to pick dependencies carefully now are left with no choice to use old-school behaviour.